torch/_inductor/scheduler.py

```diff
@@ -1845,6 +1865,13 @@ def _init(self, nodes: List[ir.Operation]) -> None:
         if config._pre_fusion_custom_pass is not None:
             self.nodes = config._pre_fusion_custom_pass(self.nodes)
         self.nodes = self.fuse_nodes(self.nodes)
```

Comment:
Hmm, I understand we want to do some amount of fusion prior to fusion of split nodes, however, the current fusion occurs so that we greedily fuse most profitable fusions first (saved global memory). And we are currently willing to do it for 10 iterations or until we did not do any fusions in the current round.

I'm not sure what the best approach is here but I don't think we want to do 20 total rounds. Potentially some smaller number of possible fusions, first, then split reductions, then fuse. And in first round we could consider just attempting fusing pointwise nodes.